### ISSUE TITLE
Let'sEncrypt - certificate for Webmin/Usermin

### ISSUE THREAD

**Issue by qooob** [webmin/webmin#520]
Jamie, hi.

Webmin/Usermin SSL setup is loading the certificate from certain, custom location, which _**doesn't**_ point to `/etc/(webmin|usermin)` directory. 

It's very important (at least for me) that it's considered. I don't want anything to be changed from the way I designed it on the system. It should be respected, and all Webmin/Usermin should do, is to overwrite certificates at the existing location. I don't want anything to be changed back its defaults, to `/etc/(webmin|usermin)/my_domain_cert.(key|cert)`. 

In other words, I setup Webmin/Usermin to read needed certificates from other directory, in my case it's `/etc/pki/` and I want this to be respected by Virtualmin Let'sEncrypt.

I'd be very appreciated if this could be done in the next 1.840 release.

**Comment by jcameron** [comment]
So do you mean that when a Let's Encrypt cert is copied to webmin or usermin, the existing files under /etc/pki be used instead of the cert being copied to /etc/webmin ?

The reason it works the way it does currently is that I don't want to overwrite a cert that the user created outside of webmin and placed in a different directory.

**Comment by qooob** [comment]
Yes, absolutely.  But only if a user explicitly points to the certificates that is outside of Webmin/Usermin configuration folder.

In my case scenario all LetsEncrypt in Virtualmin should do is just to update the certificates (whereever they are). Right now, it copies them to the `/etc/(webmin|usermin)` folders and changes miniserv.conf, with the new paths to certificates. Very annoying, as it's not how it was designed at the first place.

**Comment by jcameron** [comment]
I suppose for renewals, we could just copy the cert over the same location since clearly the user doesn't want to preserve the old one. However, I need to be *very* careful with this kind of behavior, as some users might want to keep the old cert and will get angry if we over-write it.

Maybe in your case, you could just make the .pem files in /etc/webmin symlinks to your preferred location?

**Comment by qooob** [comment]
Jamie, let me explain. I feel you don't understand what I mean.

Webmin reads but **doesn't** issues the certificates. It's all done in Virtualmin.

Webmin configuration has now in SSL settings:

**Private key file:** `/etc/pki/virtual-hosts/mydomain.com.key`
**Certificate files:** `/etc/pki/virtual-hosts/mydomain.com.cert`
**Additional CA files:** `/etc/pki/virtual-hosts/mydomain.com.ca`

In Webmin, on the same page in the tab _**Let's Encrypt**_ there is **nothing** there in setup.

At the same time in Virtualmin in domain `mydomain.com` there are all settings set properly and scheduled.

In case the certificate is renewed (manually or automatically), **_it's not only_** updating the targets mention above in `/etc/pki/..` but also **_reconfigures_** Webmin, copies the newly issued certificate/key to `/etc/(webmin|usermin)` and changes paths accordingly in `minserv.conf`.

This is what is completely wrong. It mustn't happen, right?

**Comment by jcameron** [comment]
Right, I see what you mean. The reason it was done that way is that by default certs are under domains' home directories, and if webmin read them directly it could potentially be broken by a domain owner deleting or editing the files.

However, in your case you're using /etc/pki so you're safe from this. I will change the behaviour for setups like yours in the next release.

**Comment by qooob** [comment]
Thanks, Jamie!
